utest/output/test_pylogging.py

```diff
@@ -0,0 +1,141 @@
+import unittest
+
+from robot.utils.asserts import assert_equal, assert_true, assert_false
+
+import robot.output.pyloggingconf as pyLogging
+
+import logging
+
+
+class MessageMock:
+    def __init__(self, timestamp, level, message):
+        self.timestamp = timestamp
+        self.level = level
+        self.message = message
+
+
+class MockLibraryLogger:
+    def __init__(self):
+        self.last_message = (None, None)
+
+    def error(self, message):
+        self.last_message = (message, logging.ERROR)
+
+    def warn(self, message):
+        self.last_message = (message, logging.WARNING)
+
+    def info(self, message):
+        self.last_message = (message, logging.INFO)
+
+    def debug(self, message):
+        self.last_message = (message, logging.DEBUG)
+
+    def trace(self, message):
+        self.last_message = (message, logging.NOTSET)
+
+
+class TestPyLogging(unittest.TestCase):
+    def setUp(self):
+        self.library_logger = MockLibraryLogger()
+        self.test_handler = pyLogging.RobotHandler(library_logger=self.library_logger)
+        root = logging.getLogger()
+        root.setLevel(logging.NOTSET)
+        for handler in root.handlers:
+            root.removeHandler(handler)
+        root.addHandler(self.test_handler)
+
+    def tearDown(self):
+        root = logging.getLogger()
+        root.removeHandler(self.test_handler)
+
+    def test_python_logging_debug(self):
+        TESTMESSAGE = "This is a test message"
+        logging.debug(TESTMESSAGE)
+        message, level = self.library_logger.last_message
+        assert_equal(message, TESTMESSAGE)
+        assert_equal(level, logging.DEBUG)
```

Comment:
These last three steps seem to be repeated with most if not all tests. Perhaps it would be a good idea to add a helper like this to avoid duplication:
```python
def assert_message(self, message, level):
    assert_equal(self.library_logger.last_message, message)
    assert_equal(self.library_logger.last_level, level)
```
Notice that the above would require changing `MockLibraryLogger` to store `last_message` and `last_level` separately.

Reply:
Another alternative would be enhancing `MockLibraryLogger` so that it would get something like
```python
def assert_last(self, message, level):
    assert_equal(self.last_message, message)
    assert_equal(self.last_level, level)
```
that could then me used like `self.library_logger.assert_last(...)` in tests.